solutions/sum_numbers.py

```diff
@@ -0,0 +1,61 @@
+""" sum numbers 
+sum numbers is a recursive function return the sum of count down numbers 
+
+created: 25/12/2024
+Team number:28
+Team name:MIT Alpha
+The author: Maab Mohamedkhair"""
+
+
+def sum_numbers(n):
+    
+    """ sum numbers 
+sum numbers is a recursive function return the sum of count down numbers 
+from n -> 1
+
+Base case: n == 1 
+    return 1
+
+recursion case: for n > 1
+    smaller_result of n + the value of sum_numbers(n - 1)
+
+parameters: n = any positive integer less than 1000 because python limitation in recursion 
+
+return : the sum of all numbers less than n inclusive and greater than or equal one
+
+Examples:
+
+>>> sum_numbers(4)
+10
+
+>>> sum_numbers(5)
+15
+
+>>> sum_numbers(0)
+Traceback (most recent call last):
+...
+Exception: Input must be greater than 0
+"""
+
+
+    #checks n is less than 1000
+    if (n>1000):
+        raise RecursionError
```

Comment:
this should be a defensive assertion with a raise string
assert n<1000 , "Recursion Error"

Reply:
Thanks for your important instructions 

I didn't understand what you meant exactly here.
 Why do you suggest an assertion error while the error type is already specified as a recursion error?

Reply:
using assertion is to verify conditions that should never fail in normal operation like for example n should never be larger than 1000.
it is a better implementation for debugging and development than using if statements

Reply:
Ok I get it now